### DISCUSSION TITLE
comment utilser les resultas de ce model hors connectiom

### DISCUSSION THREAD

**Discussion by Tkbg237** [roboflow/supervision#1328] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

j'ai projet sur le quel je travail qui consiste a calculer les dimensions reelles en mm d'un oeuf a partir de son image et je voulais savoir comment integrer ce model a mon projet et comment utiliser ses resultats pour avoir les dimensions en pixels de l'oeuf sur l'image

**Comment by LinasKo** [comment]
Salut @Tkbg237 👋 ,

Dans votre situation, j'utiliserais `inference` ou `ultralytics`. Voici un exemple de comment vous pouvez exécuter un modèle et obtenir le résultat : https://supervision.roboflow.com/latest/how_to/detect_and_annotate/#run-detection

Les valeurs des pixels sont obtenues en appelant `detections.xyxy`.

Pour une utilisation hors ligne, je vous suggère de lire ceci https://inference.roboflow.com/models/from_local_weights/ pour `inference` ou d'utiliser `ultralytics`.

Aussi, nous répondons généralement plus rapidement si vous pouvez écrire en anglais. Vous pouvez utiliser ChatGPT ou Google Translate. 😉

**Reply by Tkbg237** [reply]
Thanks for your help i will try the solution

**Reply by Tkbg237** [reply]
@LinasKo .  The links you provide to me are not reachable

**Reply by LinasKo** [reply]
Interesting; they work for me.

The information can be found in:
* https://supervision.roboflow.com/
* https://inference.roboflow.com/

In the first, go to "Learn".

In the second, go to LocalWeights on the left side.

**Reply by Tkbg237** [reply]
@LinasKo Can you please save it in a Google Drive document and then provide the link here please !!

**Reply by LinasKo** [reply]
https://docs.google.com/document/d/17xJxO5hx248jmOefxdt_ObzwyY14CX1wBADYfkPnNGQ/edit?usp=sharing

If that does not help, I'm afraid I can't help much more. Though here's some code to get you started:

```python
import cv2
import supervision as sv
from ultralytics import YOLO

model = YOLO("yolov8n.pt")
image = cv2.imread(<SOURCE_IMAGE_PATH>)
results = model(image)[0]
detections = sv.Detections.from_ultralytics(results)
```

**Reply by Tkbg237** [reply]
@LinasKo i would like to know how to get Real measure of an objet from his image

**Reply by LinasKo** [reply]
Easiest way: if the camera is always at the same place, and objects will always be at the same distance, measure an example object and then compare its size to what the model detects.

If that can't be done, it becomes an advanced computer vision problem and is beyond what I can help with.

**Reply by Tkbg237** [reply]
@LinasKo because i'm facing this problem in m'y projet and the solution that i wanted to applie IS use a model Size in pixels and calculate a ratio witch will be used to convert the pixels Size to Real Size what do you think?

**Reply by LinasKo** [reply]
Good. If the camera does not move, this is the easiest solution.

**Reply by Tkbg237** [reply]
@LinasKo camera doesn't move because i don't camera directly but i use a pic already Taken !!
